### DISCUSSION TITLE
Describe the CSV fields in more detail

### DISCUSSION THREAD

**Discussion by ddennerline3** [ludeknovy/jtl-reporter#294] [General]
Would it be possible to improve the documentation for the JTL reporter upload file CSV fields please?

https://jtlreporter.site/docs/integrations/introduction

- Why are the field names in documentation different between the plugin, listener, and be DB model - case (Connect vs. connect)?
- Can the same name be used across all services (file, API, and DB)?
- What is the difference between connect, latency, and response time?
- Are all the time related (connect, latency, idle time and response time) metrics used in reporting?

**Comment by ludeknovy** [comment]
hi @ddennerline3 
Regarding the csv format and its explanation, see jmeter documentation: https://jmeter.apache.org/usermanual/listeners.html#csvlogformat

> Why are the field names in documentation different between the plugin, listener, and be DB model - case (Connect vs. connect)?

The project started with CSV parsing; later on I've added the other options like a listener service, which uses a slightly different interface. Is this an issue for you?

> Can the same name be used across all services (file, API, and DB)?

Not really, the CSV parsing required the defined column names with capital letters for some of them to work properly.

> What is the difference between connect, latency, and response time?

Start here for a basic explanation: https://jmeter.apache.org/usermanual/listeners.html#csvlogformat
and maybe this could be helpful as well? https://jtlreporter.site/blog/2023/01/09/performance-metrics

> Are all the time related (connect, latency, idle time and response time) metrics used in reporting?

Yes, but it also depends on the tool used: Some metrics won't be available with Locust.io as it does not provide them, eg: latency, connection time and others.

**Reply by ddennerline3** [reply]
A slight improvement in documentation would be helpful when trying to identify why client code isn't working. An example of this is that the v2 jtl-listener-service seems to have been retired and API variable names have changed.

I am trying to build a new locust jtl-reporter plugin, and it's been non-trivial to find out the source of truth for using the streaming API.

![image](https://github.com/user-attachments/assets/90de648d-24af-4604-b334-15ba4a6c0991)

**Reply by ludeknovy** [reply]
ahh, I see in the docs, the v1 is referenced https://jtlreporter.site/docs/integrations/samples-streaming
The actual implementation of jtl listener service can be found here: https://github.com/ludeknovy/jtl-reporter-listener-service/tree/main/src/routes and the json schemas validated by the server are here: https://github.com/ludeknovy/jtl-reporter-listener-service/blob/main/src/jsonSchema/saveDataBodySchema.ts

Hope this helps.

**Reply by ludeknovy** [reply]
https://github.com/ludeknovy/jtl-reporter-docs/pull/53
